server/src/main/java/org/elasticsearch/common/path/PathTrie.java

```diff
@@ -50,46 +52,39 @@ enum TrieMatchingMode {
         TrieMatchingMode.WILDCARD_ROOT_NODES_ALLOWED
     );
 
-    public interface Decoder {
-        String decode(String value);
-    }
-
-    private final Decoder decoder;
+    private final UnaryOperator<String> decoder;
     private final TrieNode root;
     private T rootValue;
 
     private static final String SEPARATOR = "/";
     private static final String WILDCARD = "*";
 
-    public PathTrie(Decoder decoder) {
+    public PathTrie(UnaryOperator<String> decoder) {
         this.decoder = decoder;
-        root = new TrieNode(SEPARATOR, null, WILDCARD);
+        root = new TrieNode(SEPARATOR, null);
     }
 
-    public class TrieNode {
-        private transient String key;
-        private transient T value;
-        private final String wildcard;
-
-        private transient String namedWildcard;
+    private class TrieNode {
+        private String key;
+        private T value;
+        private String namedWildcard;
 
         private Map<String, TrieNode> children;
 
-        private TrieNode(String key, T value, String wildcard) {
+        private TrieNode(String key, T value) {
             this.key = key;
-            this.wildcard = wildcard;
             this.value = value;
             this.children = emptyMap();
             if (isNamedWildcard(key)) {
-                namedWildcard = key.substring(key.indexOf('{') + 1, key.indexOf('}'));
+                namedWildcard = key.substring(1, key.length() - 1);
```

Comment:
turned out that in the past a wildcard could be in the middle of the path part? I don't think we use this in our codebase though
example /abc{wildcard}abc/
This is ok by me, do we want to have a 'negative' test case to prevent the wildcard in the middle?

Reply:
I'm not sure why we would need wildcards in the middle of a key, its been like this since initial commit in 2010. I don't think this is something we would ever actually do anyway. The fact that everything is still working shows that it isn't needed.

Reply:
I agree we don't need this, it surprised me that the code allowed it (could plugins use it this way?)
it would be a very low level breaking change to me

Reply:
I suppose plugins can declare routes too. However, given there are no tests for this functionality, it maybe is incidental that it could do this?

Reply:
Turns out partial matches dont actually work. So it's all good